tests/python_package_test/test_basic.py

```diff
@@ -654,16 +654,18 @@ def test_no_copy_when_single_float_dtype_dataframe(dtype, feature_name):
     assert np.shares_memory(X, built_data)
 
 
-@pytest.mark.parametrize('feature_name', [['x1'], 'auto'])
+@pytest.mark.parametrize('feature_name', [[42], 'auto'])
```

Comment:
```suggestion
@pytest.mark.parametrize('feature_name', [['x1'], [42], 'auto'])
```

Instead of replacing a string with a number here, would you consider *adding* a case here? That way, this test would still cover the case "`feature_name` provided, and names are strings".

Reply:
Thanks for the proposal! Added in 2b107f6d883f0931456cbdb973fbe0ef626d0f0d.

Reply:
changes look great, thank you!